astropy/config/configuration.py

```diff
@@ -366,7 +366,7 @@ def set(self, value):
             raise TypeError(
                 f"Provided value for configuration item {self.name} not valid:"
                 f" {e.args[0]}"
-            )
+            ) from e
```

Comment:
While I see why you did it for the other lines (so far), I don't see that for this one. Why did you decide to add it here?

Thanks!

Reply:
Thanks for the clarification. @Cadair and @saimn might have opinions when it comes to config.